examples/nextjs/Dockerfile

```diff
@@ -0,0 +1,61 @@
+# Source: https://github.com/vercel/next.js/blob/canary/examples/with-docker/Dockerfile
+
+FROM node:18-alpine AS base
+
+# Install dependencies only when needed
+FROM base AS deps
+# Check https://github.com/nodejs/docker-node/tree/b4117f9333da4138b03a546ec926ef50a31506c3#nodealpine to understand why libc6-compat might be needed.
+RUN apk add --no-cache libc6-compat
+WORKDIR /app
+
+# Install dependencies based on the preferred package manager
+COPY package.json docker-package-lock.json ./
```

Comment:
I'm struggling to understand what the `docker-package-lock.json` file is for, how it's supposed to get updated, etc.